### PR TITLE
Add a `--local` argument to install.sh

### PR DESCRIPTION
**Pull request by murez**
add option `--local` to install at `$HOME/.local` and check permission of the prefix before installing

### CODE REVIEW

File: script/install.sh [hunk 1]
```diff
@@ -9,14 +9,37 @@ elif [ "${BOXEN_HOME:-}" != "" ] ; then
...
 fi
 
-mkdir -p $prefix/bin
-rm -rf $prefix/bin/git-lfs*
+local_install=false
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --local)
+      local_install=true
```
Comment (bk2204) [inline script/install.sh:17]: It doesn't look like this variable is being used.  Perhaps we can remove it here and on line 12?

File: script/install.sh [hunk 2]
```diff
@@ -9,14 +9,37 @@ elif [ "${BOXEN_HOME:-}" != "" ] ; then
...
+    *)
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
+done
+
+# Check if the user has permission to install in the specified prefix
+if [ ! -w "$prefix" ]; then
+  echo "Error: Insufficient permissions to install in $prefix. Try running with sudo or choose a different prefix."
```
Comment (bk2204) [inline script/install.sh:30]: Let's redirect this to standard error with `>&2` so that errors go the right place.

**Review by bk2204 — COMMENTED** [review]
Hey, thanks for the patch and welcome to Git LFS!  This looks like a valuable feature.  I have just a few things that we could clean up here and then this should be ready to go.

**Comment by murez** [comment]
Thanks for suggestions, I just found out that on multi tenant like HPC env, it is impossible to install `git-lfs` to `/usr/local`, and install to home local may be the best practice.

**Comment by bk2204** [comment]
It's not actually impossible to do.  You need to specify the `PREFIX` environment variable, and if you do `PREFIX="$HOME/.local" ./install.sh`, then it should work.  That's a pretty standard way to specify installation targets for build systems, but it would be nice to have support for this out of the box.

**Review by bk2204 — APPROVED** [review]

**Pull request merged**